app/channels/application_cable/connection.rb

```diff
@@ -1,4 +1,20 @@
 module ApplicationCable
   class Connection < ActionCable::Connection::Base
+    identified_by :current_user
+
+    def connect
+      self.current_user = find_verified_user
+      logger.add_tags 'ActionCable', current_user.email
+    end
+
+    private
+    def find_verified_user 
+      if verified_user = User.find_by(id: cookies[:user_id])
```

Comment:
When you need to find an entity by their primary key (id), you can take advantage of `find` method. You will not need to specify the key, since rails automatically knows that in that case your intention is to find a record that matches with the id provided. In this case, it will be:

```
User.find(cookies[:user_id]))

instead of

User.find_by(id: cookies[:user_id])
```

Reply:
The problem with find is that it will return an exception if the element is not found. find_by returns nil.

Reply:
I got what you mean. `find_by` will be good then. 👍